src/Executor/Promise/Adapter/SyncPromiseAdapter.php

```diff
@@ -24,19 +23,19 @@ class SyncPromiseAdapter implements PromiseAdapter
      */
     public function isThenable($value)
     {
-        return $value instanceof Deferred;
+        return $value instanceof SyncPromise;
     }
 
     /**
      * @inheritdoc
      */
     public function convertThenable($thenable)
     {
-        if (! $thenable instanceof Deferred) {
+        if (! $thenable instanceof SyncPromise) {
             throw new InvariantViolation('Expected instance of GraphQL\Deferred, got ' . Utils::printSafe($thenable));
```

Comment:
```suggestion
            throw new InvariantViolation('Expected instance of ' . SyncPromise::class . ', got ' . Utils::printSafe($thenable));
```

Reply:
True 👍, also using sprintf would increase readability 🤔

Reply:
I intensily dislike think `sprintf`, i think it makes it hard to know what goes into a message, is prone to errors and makes the code brittle and harder to change.

Reply:
We don't expose `SyncPromise` in docs or anywhere in error messages because it may lead to confusion.

If you create `SyncPromise` without an `executor` function in the constructor and return from your resolver - it will never resolve (actually throw while trying). The whole point of `Deferred` was to ensure it never happens and that any resolver creates at least one `executor` to start the promise chain.

So I just don't want to expose `SyncPromise` to users (in a hope to avoid confusion).

Reply:
I can get behind that reasoning, how about we put that into the DocBlock of the `SyncPromise` class? A one-liner above this error message would also be helpful.

Reply:
Yep, will do when I have a minute, thanks!